src/asar.ts

```diff
@@ -33,6 +33,71 @@ function isUnpackedDir(dirPath: string, pattern: string, unpackDirs: string[]) {
   }
 }
 
+export type FileProperties = {
+  filepath: string;
+  properties: {
+    unpack?: boolean;
+  };
+};
+
+async function getFileOrdering(
+  options: CreateOptions,
+  src: string,
+  filenames: string[],
+): Promise<FileProperties[]> {
+  if (!options.ordering) {
+    return filenames.map<FileProperties>((filepath) => ({ filepath, properties: {} }));
+  }
+  const orderingMap: FileProperties[] = (await fs.readFile(options.ordering))
+    .toString()
+    .split('\n')
+    .map<FileProperties>((line) => {
+      line = line.trim();
+      const config: FileProperties = { filepath: line, properties: {} };
+      const colonIndex = line.indexOf(':');
+      if (colonIndex > -1) {
+        config.filepath = line.substring(0, colonIndex); // file path
+        const props = line.substring(colonIndex + 1); // props on other side of the `:`
+        config.properties = props.length > 2 ? JSON.parse(props) : {}; // file properties
+      }
+      if (config.filepath.startsWith('/')) {
+        config.filepath = config.filepath.slice(1);
+      }
+      return config;
+    });
+
+  const ordering: FileProperties[] = [];
+  for (const config of orderingMap) {
+    const pathComponents = config.filepath.split(path.sep);
+    let str = src;
+    for (const pathComponent of pathComponents) {
+      str = path.join(str, pathComponent);
+      ordering.push({ filepath: str, properties: config.properties });
+    }
+  }
+
+  let missing = 0;
+  const total = filenames.length;
+
+  const fileOrderingSorted: FileProperties[] = [];
+
+  ordering
+    .filter((config) => !fileOrderingSorted.includes(config) && filenames.includes(config.filepath))
+    .forEach((config) => {
+      fileOrderingSorted.push(config);
+    });
```

Comment:
Just realized this `filter=>foreach` needs to be a `for-of`. Will push this fix asap